extensions/advanced_ballistics/AdvancedBallistics.cpp

```diff
@@ -309,7 +249,7 @@ double calculateZeroAngle(double zeroRange, double muzzleVelocity, double boreHe
 
             v = std::sqrt(vx*vx + vy*vy);
 
-            double retard = calculateRetard(dragModel, ballisticCoefficient, v);
+            double retard = calculateRetard(dragModel, ballisticCoefficient, v, SPEED_OF_SOUND(15));
```

Comment:
You are calculating this constant.. at runtime. Everytime the function is called. Why?

Reply:
RELEVANT VARIABLE NAME

Reply:
@dedmen, I expect the compiler to optimize it out.

Reply:
It won't be here. because std::sqrt is not constexpr.

Reply:
gcc 7.2.0 is smart enough. I will test it with VS 2017 tomorrow.

Reply:
Visual Studio 2017 produces the exact same assembly code in both cases:
```
#define SPEED_OF_SOUND(t) (331.3 + std::sqrt(1 + t / 273.15f))
...
double r = calculateRetard(dragModel, ballisticCoefficient, v, SPEED_OF_SOUND(15));
```
```
#define SPEED_OF_SOUND(t) (331.3 + std::sqrt(1 + t / 273.15f))
...
const double c = SPEED_OF_SOUND(15);
double r = calculateRetard(dragModel, ballisticCoefficient, v, c);
```

Reply:
moving it into a const float doesn't change that std::sqrt is not constexpr before C++17.
Actually.. Not even C++17 has constexpr sqrt.
Does that assembly code that it produces calculate the Speed of sound. Or does it use a constant? If it calculates then it does exactly what I said

Reply:
Just did a release compile. And yes it is calling sqrt at runtime.

Reply:
Ok, suggestions?